packages/redux/src/analytics/middlewares/types/bag.types.ts

```diff
@@ -0,0 +1,27 @@
+import type {
+  BagItemEntity,
+  MerchantEntity,
+  ProductEntity,
+} from '../../../entities/types';
+
+export type BagProductActionMetadata = {
+  authCode: string;
+  bagId: string;
+  bagItemId: BagItemEntity['id'];
+  customAttributes: string;
+  from: string;
+  merchantId: MerchantEntity['id'];
+  productAggregatorId: number;
+  productId: ProductEntity['id'];
+  quantity: BagItemEntity['quantity'];
+  scale: number;
+  size: number;
+};
+
+export type BagActionMiddlewareOptions = {
+  ADD_BAG_ITEM_SUCCESS: string;
+  UPDATE_BAG_ITEM_SUCCESS: string;
+  REMOVE_BAG_ITEM_SUCCESS: string;
+};
+
+export type BagActionCustomOptions = Partial<BagActionMiddlewareOptions>;
```

Comment:
I think I would change these a little bit. `BagActionCustomOptions` I would rename to `BagActionMiddlewareOptions` as this is what the developer will see and `BagActionMiddlewareOptions` I would rename to `BagActionMiddleware<Merged|Final|Processed>Options`.